@here/harp-texture-atlas-generator/src/FileSystem.ts

```diff
@@ -0,0 +1,284 @@
+     * specified.
+     */
+    static getFilesListWithExtensions(directoryPath: string, fileExtensions: string[]): string[] {
+        const files: string[] = fileSystem.readdirSync(directoryPath);
+        const filesFiltered = files.filter(fileName => {
+            for (const fileExt in fileExtensions) {
+                if (fileName.toLowerCase().endsWith(fileExt)) {
+                    return true;
+                }
+            }
+            return false;
+        });
+        return filesFiltered;
+    }
+
+    /**
+     * Find list of files (images) in directory matching the format specified.
+     *
+     * @param directoryPath absolute path to the directory.
+     * @param imageFormat image format to be searched for.
+     * @note Method uses naive approach that is based on the file extension, it
+     * will never recognize files without known image extension.
+     */
+    static getFilesListWithFormat(directoryPath: string, imageFormat: ImageFormat): string[] {
+        const fileExtensions: string[] = FileSystem.getImageFormatExtensions(imageFormat);
+        return FileSystem.getFilesListWithExtensions(directoryPath, fileExtensions);
+    }
+
+    /**
+     * Synchronously reads data buffer from file.
+     *
+     * @param path A path to a file. If a URL is provided, it must use the `file:`
+     * protocol.
+     * @returns file contents in Buffer object.
+     */
+    static readFileSync(filePath: string): Buffer {
+        return fileSystem.readFileSync(filePath);
+    }
+
+    /**
+     * Read entire content of the file asynchronously.
+     *
+     * @param filePath A path to a file. If a URL is provided, it must use the `file:`
+     * protocol.
+     * @returns Promise with file contents in Buffer object.
+     */
+    static readFile(filePath: string): Promise<Buffer> {
+        const promise = new Promise<Buffer>((resolve, reject) => {
+            fileSystem.readFile(filePath, {},
+                (error: NodeJS.ErrnoException, buffer: Buffer) => {
+                    if (error) {
+                        reject(new Error(error.message));
+                    }
+                    else {
+                        resolve(buffer);
+                    }
+                });
+        });
+        return promise;
+    }
+
+    /**
+     * Write entire buffer to file synchronously, replacing the file if it already exists.
+     *
+     * @param filePath file storage path.
+     * @param data buffer to be stored.
+     */
+    static writeFileSync(filePath: string, data: Buffer | string) {
+        fileSystem.writeFileSync(filePath, data);
+    }
+
+    /**
+     * Write entire buffer to file asynchronously, replacing the file if it already exists.
+     *
+     * @param filePath file storage path.
+     * @param data buffer to be stored.
+     * @returns Promise.
+     */
+    static writeFile(filePath: string, data: Buffer | string): Promise<any> {
```

Comment:
Return type should be `Promise<void>`
You can use `util.promisify` for boilerplate like this.

If you would use `namespace` for FileSystem as proposed in other comment, it would be as simple as.

```
export const writeFile = util.promisify(fs.writeFile);
```

some marketing: https://hackernoon.com/node8s-util-promisify-is-so-freakin-awesome-1d90c184bf44
ref: https://nodejs.org/api/util.html#util_util_promisify_original